pkg/providers/instancetype/types.go

```diff
@@ -137,6 +141,38 @@ func computeRequirements(ctx context.Context, info *ec2.InstanceTypeInfo, offeri
 	return requirements
 }
 
+func getOS(info *ec2.InstanceTypeInfo) []string {
+	if isSupportedForWindows(info) {
+		return []string{string(v1.Linux), string(v1.Windows)}
+	}
+	return []string{string(v1.Linux)}
+
+}
+
+func isSupportedForWindows(info *ec2.InstanceTypeInfo) bool {
+
+	//Explicitly supported for windows
+	if lo.Contains(instanceTypesSupportedForWindows, *info.InstanceType) {
+		return true
+	}
+
+	//Non amd64
+	if getArchitecture(info) != v1alpha5.ArchitectureAmd64 {
+		return false
+	}
+
+	//https://docs.aws.amazon.com/eks/latest/userguide/windows-support.html
+	//Amazon EC2 instance types C3, C4, D2, I2, M4 (excluding m4.16xlarge), M6a.x, and R3 instances aren't supported for Windows workloads.
+	//From AWS support, M6a.x means all types in m6a family
+	split := strings.Split(*info.InstanceType, ".")
```

Comment:
Is it possible to create a generic function for getting the instance type parts since we do it once here and we also do it again in `computeRequirements`?